userspace/falco/app/actions/helpers_interesting_sets.cpp

```diff
@@ -75,14 +88,45 @@ static void select_event_set(falco::app::state& s, const libsinsp::events::set<p
 	auto base_event_set = libsinsp::events::sinsp_state_event_set();
 	s.selected_event_set = rules_event_set.merge(base_event_set);
 
+	auto user_base_syscalls_names = s.config->m_base_syscalls;
+	if (!user_base_syscalls_names.empty())
+	{
+		auto valid_events = libsinsp::events::names_to_event_set(user_base_syscalls_names);
```

Comment:
The way we get `valid_events` is the only thing that I'm not sure about.

- We should pass down to `libsinsp::events::names_to_event_set` a list of names without the `!xxx` negative ones. Even if the API ignores unknown names, I still think it's not a good practice
- If `base_syscall` is not empty, but contains only negative syscall names, then `valid_events` will be empty, which would cause the selected set to contain only the ruleset-defines sycalls. I don't think this is what we want.

To fix both, I think `extract_negative_base_syscalls_names` should be generalized to split `base_syscall` in two disjunct sets of positive and negative syscalls names. Then, the positive names are used to determine `valid_events`, and if empty will fallback to `libsinsp::events::sinsp_state_event_set()`. Then, all negative syscalls will be removed at the end by `s.selected_event_set` (like you already do).

What do you think?